### PR TITLE
UnfavouriteThis Use Case Implemented

### PR DESCRIPTION
**Pull request by ryanzyi**
The unfavouriteThis use case has been implemented. Each recipe saved in the user's favouriteRecipes entity can now be removed via the favourites view. 

Preliminary testing via using the program shows that it is correctly updating the entities, states, and more. Existing code and the new code in this pull request support the edge cases where the number of recipes in favouriteRecipes is one and zero. Will complete formal testing in the next day or two.

### CODE REVIEW

File: src/use_case/unfavouriteThis/UnfavouriteThisInputData.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        this.title = title;
+        this.summary = summary;
+        this.recipeImageURL = recipeImageURL;
+        this.servings = servings;
+        this.readyInMinutes = readyInMinutes;
+        this.extendedIngredients = extendedIngredients;
+        this.extendedInstructions = extendedInstructions;
+        this.id = id;
+    }
```
Comment (HC123abc) [inline src/use_case/unfavouriteThis/UnfavouriteThisInputData.java:25]: Needs to contain the map

Comment (ryanzyi) [reply src/use_case/unfavouriteThis/UnfavouriteThisInputData.java]: fixed 😄

**Comment by ryanzyi** [comment]
Changes have been made to not make unnecessary api calls. Should be good now 👍

**Comment by HC123abc** [comment]
Nice!
-Everything else looks good
-correct convention
-follow solid and CA

**Pull request merged**